x/xio/reader_slice_of_slices_from_segment_readers_iterator.go

```diff
@@ -46,18 +48,26 @@ func (it *readerSliceOfSlicesIterator) Next() bool {
 	return true
 }
 
+func (it *readerSliceOfSlicesIterator) CurrentStart() time.Time {
+	return it.blocks[it.arrayIdx()][0].Start()
+}
+
+func (it *readerSliceOfSlicesIterator) CurrentEnd() time.Time {
+	return it.blocks[it.arrayIdx()][0].End()
```

Comment:
It might definitely be worth checking if `len(...) > 0` and if not return just `time.Time{}` here to be safe.